### ISSUE TITLE
NAOMI - Melty Blood Act Cadenza/Actress Again crashes/performance issues on Retroarch android 

### ISSUE THREAD

**Issue by DooMGuy87** [libretro/flycast#1013]
As the title says, flycast is having some issues after the last updates. Previously, both games were running perfectly (aside some graphical glitches here and there, nothing gamebreaking)
Firstly, MBAC is having some random crashes:  in example, when getting to stage 5, there is a dialog between characters, when it starts, the game crashes. Another crash occurs while fighting against certain characters, IE Satsuki.
MBAC works well, but the only problem (aside some graphical glitches) is the opening video has some lag, its strange because it played just fine previously on the same same device.
My android device is a Samsung A30S, and as I said, it's strange because both games worked perfectly on wihout any crashes/lag. Also tested the core on Retroarch, Retroarch Plus on different settings and I'm having the same result.
I'm using the MAME romsets, meltyb.zip + CHD image, and mbaa.zip.
Thanks in advance.

**Comment by kage2051** [comment]
Cannot reproduce both issues on my end. Made sure to have the latest build of the core first in case you ask.

Can you post a video showing the bug? A save state before the fifth stage would help as well.

**Comment by DooMGuy87** [comment]
Thanks for your reply!
I couldn't reproduce the 5th stage crash again (It may have been just a random one) It happened while playing Arcueid's story.
Now, i could reproduce the satzuki one. I realized that it wasn't the character since i fought her when testing Arcueid's story, but when having a fight between Akiha Vermillon Vs Satsuki, hangs the emulator after the versus screen and I have to force it close. Usually you fight her on the first fight, but I can upload a state if you want

**Comment by kage2051** [comment]
Yes, I'd like to test it. I played with Sion (one of the female characters) where I couldn't reproduce the bugs you mentioned.

EDIT: So I got to play two matches vs Satsuki without crashes of any kind. I think you should reset your flycast settings or something, and if the issue persists, then it's your Exynos GPU driver causing it (my tests were done on a Snapdragon device).

**Comment by DooMGuy87** [comment]
Here is a video of what happens in a match between Satsuki vs Akiha Vermillon, after the VS screen tha emulator hangs:
https://mega.nz/file/9EA3AIYa#16oEeFF7flFo8POgO0wbqdwAWM7Ct56YjQ6IdJeV7a8
It's weird, because I hadn't happened before (I played several times arcueid's and AV story just weeks ago without any issues).
Also, it's a Retroarch Plus fresh install with the latest flycast core and default settings.
Here is a savestate before the fight in question, its for MBAC Rev B: https://mega.nz/file/VERU3bJA#sSQLj8oqX1Qct13LRTMXprkpYO0qz355_VnmjZW0rr4

**Comment by kage2051** [comment]
So I tested the B2 version, and yes I can finally reproduce the Satsuki issue. Before you ask, I'm using the nightly builds of RA with the original Core Updater routine (don't like the limitations in the Play Store version).

Standalone flycast suffers from this too with the recent builds.

**Comment by barbudreadmon** [comment]
is it reproducible when installing the following build manually ? 
https://git.libretro.com/libretro/flycast/-/jobs/90590

**Comment by kage2051** [comment]
> is it reproducible when installing the following build manually ?
> https://git.libretro.com/libretro/flycast/-/jobs/90590

I've already reported the issue to flyinghead. It happens on the latest build of standalone too, not only on the libretro core (and yes I used that 34c8a2f1 build of the latter during my tests).

**Comment by barbudreadmon** [comment]
ok, thanks for confirming it also happens on standalone, the build above is the build just before i backported the vtennis fix, it would be interesting to confirm that fix is the cause of this issue (that's pretty much the only naomi-specific backport i did recently)

**Comment by DooMGuy87** [comment]
I tried that version and same thing happens.
On a side note, the Opening movie lag happens on MBAC REV B2 too, and a bit worse than MBAA. I know is not that important, but the thing is that they played fine before but they lag now after some seconds of playing

**Comment by flyinghead** [comment]
The vs. Satsuki hang doesn't look like a recent regression. I was able to reproduce it as far as back as June. It seems to happen randomly.

**Comment by SonofLilith49** [comment]
Hi how are things.
Does anyone have the download link for the Melty Blood Actress Again game, can you please pass me on?
It is for Retroarch Plus emulator

**Comment by fgenjoyer** [comment]
Yo its been 2 years, any solution yet? Im also experiencing the same problem, i tried everything but no luck
